rockPaperScissors.py

```diff
@@ -3,48 +3,51 @@
 
 import random
 
-def gameLogic(player, computer, playerScore, computerScore):
-    playerIndex = RPS.index(player)
-    computerIndex = RPS.index(computer)
-
-    gameIncrement = 0
-    if (playerIndex != computerIndex):
-        gameIncrement = 1
-        playerNextIndex = (playerIndex + 1) % len(RPS)
-        
-        if (playerNextIndex == computerIndex):
-            computerScore += 1
-        else:
-            playerScore += 1
+def gameLogic(player1Choice, player2Choice):
+    if player1Choice == player2Choice:
+        return 0
 
-    return gameIncrement, playerScore, computerScore
+    playerNextIndex = (player1Choice + 1) % 3
+    
+    winner = 2 if playerNextIndex == player2Choice else 1
+    
+    return winner
 
-maxGames=3
-RPS=['rock','paper','scissors']     
 
-computerScore=0   
-playerScore=0  
-gameCount=0   
+RPS = ['rock', 'paper', 'scissors']
+maxGames = 3
+computerScore = 0
+playerScore = 0
+gameCount = 0
+
 print('I challenge you to rock paper scissors, best 2 out of 3!')
-while gameCount < maxGames:    #loops for 3 rounds
-    print('rock paper scissors!')
-    player=input()
-    computer=random.choice(RPS)
-    print(computer)
-    gameIncrement, newComputerScore, newPlayerScore = gameLogic(computer, player, computerScore, playerScore)
 
-    if gameIncrement == 0:
+while gameCount < maxGames:    #loops for 3 rounds
+    playerChoice = input('rock paper scissors! Make your choice:')
```

Comment:
calling ```input``` with a string argument will print that string to the command line

Reply:
I am aware of this, I am under the impression that it saves me from having to add a line to print() the string that I want to have displayed on the screen... is that poor form?

Reply:
I apparently knew that but did not use it in this instance

Reply:
For me it’s all about clarity. In this case doing it all with one line was a little more terse (less code to have to scan through) while still being very easy to understand. That’s always the balance I look for in my code - will more or less code make this as understandable and straight forward as possible?